modules/controller/dh-controller.js

```diff
@@ -194,45 +198,122 @@ class DHController {
-            /*
-              missingObjects: {
-                id1: [  dataset 1,  dataset 2, ... ],
-                id2: [  dataset 2,  dataset x, ... ],
-                ...
-              }
-             */
+        const { identifier_types, identifier_values, opcode } = req.body;
+
+
+        try {
+            const response = await this.trailService.lookupTrail(
+                identifier_types,
+                identifier_values,
+                opcode,
+            );
 
-            const missingIds = Object.keys(missingObjects);
-            const missingElements =
-                await this.graphStorage.findTrailExtension(missingIds, missingObjects);
+            res.status(200);
+            res.send(response);
+        } catch (e) {
+            this.logger.error(e.message);
+            res.status(501);
+            res.send({ errorMessage: 'Internal error' });
+        }
+    }
 
-            const trailExtension = this.importService.packTrailData(missingElements);
+    async findTrail(req, res) {
+        this.logger.api('POST: Trail find request received.');
 
-            return response.concat(trailExtension);
+        if (req.body === undefined ||
+            req.body.unique_identifiers === undefined
+        ) {
+            res.status(400);
+            res.send({
+                message: 'Bad request',
+            });
+            return;
         }
 
-        return response;
+        const {
+            unique_identifiers, included_connection_types, excluded_connection_types,
+        } = req.body;
+
+        let { depth, reach } = req.body;
+
+        depth = depth === undefined ?
+            this.graphStorage.getDatabaseInfo().max_path_length :
+            parseInt(depth, 10);
+
+        reach = reach === undefined ?
+            constants.TRAIL_REACH_PARAMETERS.narrow : reach.toLowerCase();
+
+        const inserted_object = await Models.handler_ids.create({
+            status: 'PENDING',
+        });
+
+        const commandData = {
+            handler_id: inserted_object.dataValues.handler_id,
+            unique_identifiers,
+            depth,
+            reach,
+            included_connection_types,
+            excluded_connection_types,
+        };
+
+        await this.commandExecutor.add({
+            name: 'dhFindTrailCommand',
+            delay: 0,
+            transactional: false,
+            data: commandData,
+        });
+
+        res.status(200);
+        res.send({
+            handler_id: inserted_object.dataValues.handler_id,
+        });
+    }
+
+    async findTrailResult(req, res) {
+        const handlerId = req.params.handler_id;
+        this.logger.api(`POST: Trail result request received with handler id: ${handlerId}`);
+        const handler_object = await Models.handler_ids.findOne({
+            where: {
+                handler_id: handlerId,
+            },
+        });
+
+        if (!handler_object) {
+            this.logger.info('Invalid request');
```

Comment:
Add more descriptive log, you can use same thing as in response:
Unable to find data with given parameters! handler_id is required!

Reply:
I agree, we have to refactor logging in the project